src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java

```diff
@@ -152,6 +154,32 @@ public void testAvroWithSchemaRegistry() throws IOException {
     assert ((SnowflakeRecordContent) input.value()).getData()[0].toString().equals("{}");
   }
 
+  @Test
+  public void testAvroWithSchemaRegistryAndReaderSchema() throws IOException
+  {
+    MockSchemaRegistryClient client = new MockSchemaRegistryClient();
+    SnowflakeAvroConverter converter = new SnowflakeAvroConverter();
+    Map<String, String> configs = new HashMap<String, String>();
+    configs.put(
+            SnowflakeAvroConverter.READER_SCHEMA,
+            "{\"name\":\"test_avro\",\"type\":\"record\",\"fields\":[{\"name\":\"int\",\"type\":\"int\"},{\"name\":\"newfield\",\"type\":\"int\",\"default\": 1}]}");
+    configs.put(
+            AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+            "http://fake-url");
+    converter.configure(configs, false);
+    converter.setSchemaRegistry(client);
+
+    SchemaAndValue input = converter.toConnectData("test", client.getData());
+    SnowflakeRecordContent content = (SnowflakeRecordContent) input.value();
+    assert content.getData().length == 1;
+    assert content.getData()[0].asText().equals(mapper.readTree("{\"int" +
+            "\":1234, \"newfield\":1}").asText());
+
+    //null value
+    input = converter.toConnectData("test",null);
```

Comment:
I suppose this is the current behavior?

Reply:
Yes, I followed the same structure in the test as in the test without a reader schema. In principle, the null case could be removed here. Should I?